pkg/deployer/deployer.go

```diff
@@ -55,9 +55,18 @@ func (d *Deployer) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.M
 		return "", errors.New("key not found in status")
 	}
 
-	ip, err := d.GetIP(cluster, master)
-	if err != nil {
-		return "", err
+	var ip string
+	if master != nil {
+		ip, err = d.GetIP(cluster, master)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		// This case means no master created yet, we need get from cluster info anyway
+		if len(config.MasterIP) == 0 {
+			return "", errors.New("MasterIP in cluster spec not set")
+		}
+		ip = config.MasterIP
 	}
```

Comment:
We have to merge this with this code here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/401/files#diff-266c339b51cf18a076cf106526357c53R70-R78

Reply:
I guess this would be in the else branch if ManagedAPIServerLoadBalancer is set to `false`

Reply:
yes, let's work on that patch first then rebase from there

Reply:
Yup. I'll rebase mine